scripts/addr2line.py

```diff
@@ -54,46 +62,55 @@ def __init__(self, binary, concise=False, cmd_path="addr2line"):
             print('{}'.format(s))
 
         options = f"-{'C' if not concise else ''}fpia"
-        self._input = subprocess.Popen([cmd_path, options, "-e", self._binary], stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+        self._input_proc = subprocess.Popen([cmd_path, options, "-e", self._binary], stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
         if concise:
-            self._output = subprocess.Popen(["c++filt", "-p"], stdin=self._input.stdout, stdout=subprocess.PIPE, universal_newlines=True)
+            self._output_proc = subprocess.Popen(["c++filt", "-p"], stdin=self._input_proc.stdout, stdout=subprocess.PIPE, universal_newlines=True)
         else:
-            self._output = self._input
+            self._output_proc = self._input_proc
 
         # If a library doesn't exist in a particular path, addr2line
         # will just exit.  We need to be robust against that.  We
         # can't just wait on self._addr2line since there is no
         # guarantee on what timeout is sufficient.
-        self._input.stdin.write('\n')
-        self._input.stdin.flush()
-        res = self._output.stdout.readline()
+        self._input.write('\n')
+        self._input.flush()
+        res = self._output().readline()
         self._missing = res == ''
 
+    @property
+    def _input(self):
+        """Returns the input stream for the process/pipe."""
+        return notNone(self._input_proc.stdin)
+
+    def _output(self):
+        """Returns the output stream for the process/pipe."""
+        return notNone(self._output_proc.stdout)
+
     def _read_resolved_address(self):
-        res = self._output.stdout.readline()
+        res = self._output().readline()
         # remove the address
         res = res.split(': ', 1)[1]
         line = ''
         while Addr2Line.dummy_pattern.fullmatch(line) is None:
             res += line
-            line = self._output.stdout.readline()
+            line = self._output().readline()
         return res
 
-    def __call__(self, address):
+    def __call__(self, address: str):
         if self._missing:
             return " ".join([self._binary, address, '\n'])
         # We print a dummy 0x0 address after the address we are interested in
         # which we can look for in _read_address
-        self._input.stdin.write(address + '\n0x0\n')
-        self._input.stdin.flush()
+        self._input().write(address + '\n0x0\n')
```

Comment:
if `_input` is a property, we don't need *call* it for retrieving the value returned by it, am i right?

Reply:
Sorry, this is correctly without `()` after the 2nd commit in this series, but [absorb](https://github.com/tummychow/git-absorb) assigned it to the wrong commit in this case, let me fix that.

Reply:
Fixed in https://github.com/scylladb/seastar/commit/db9e2ff7afa96c0e533b55c5b2abbdcf755e8f46.